PacketParser.py

```diff
@@ -9,30 +9,41 @@ class PacketParser:
     def __init__(self, filename):
         self.filename = filename
         self.sections = []
-
-    def _parse_row(self, row):
+    
+    def _parse_row_tel(self, row):
+        if row[0] != "":
+            if row[-1] == "TEL":
+                self.sections.append(PacketSection(row[0], PacketType.TEL))
+            else:
+                raise ValueError("Protocol not TEL")
+        if row[1] != "":
+            if row[-1] == "TEL":
+               self.sections[-1].add_packet(TELPacket(row[1]))
+        self.sections[-1].get_last_packet().add_field(row)
+    
+    def _parse_row_can(self, row):
         if row[0] != "":
-            if row[-1] == "":
-                raise ValueError("Protocol not specified")
             if row[-1] == "CAN":
                 self.sections.append(PacketSection(row[0], PacketType.CAN))
-            elif row[-1] == "TEL":
-                self.sections.append(PacketSection(row[0], PacketType.TEL))
-        if row[1] != "":         
-            # Check if the protocol is CAN
+            else:
+                raise ValueError("Protocol not CAN")
+        if row[1] != "":
             if row[-1] == "CAN":
                 self.sections[-1].add_packet(CANPacket(row[1]))
-            elif row[-1] == "TEL":
-                self.sections[-1].add_packet(TELPacket(row[1]))
         self.sections[-1].get_last_packet().add_field(row)
-
-    def parse(self):
+    
+    def parse(self,dtype):
         with open(self.filename) as f:
             reader = csv.reader(f)
-            for row in reader:
-                if row[0] == "Section":
-                    continue
-                self._parse_row(row)
+            if dtype == "CAN":
```

Comment:
We can instead pass dtype to parse row and have parse row decide what to do. There would be less duplicate code that way